nototools/report_coverage_data.py

```diff
@@ -0,0 +1,286 @@
+    block_cps = unicode_data.block_chars(block_name)
+    block_cps &= defined_cps
+    if not block_cps:
+      continue
+    cov_info = []
+    all_empty = True
+    for covered_cps in covered_cps_list:
+      block_covered_cps = covered_cps & block_cps
+      if block_covered_cps:
+        all_empty = False
+      cov_info.append(block_covered_cps)
+    if no_empty and all_empty:
+      continue
+    block_data.append(block_range + (block_name, block_cps, cov_info))
+  return block_data
+
+
+def write_block_coverage_html(block_data, names, msg, out_file=sys.stdout):
+  cp_limit = 1000
+  chart_width = 250
+  chart_row_height = 8
+
+  HEADER = """<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="utf-8">
+  <title>Block coverage</title>
+  <style>
+  td.range {font-family: monospace; text-align: right;}
+  td.count {font-family: monospace; text-align: right;}
+  td.pct {font-family: monospace; text-align: right;}
+  div.chart {width:%dpx; position:relative;}
+  div.tot {border: 1px solid black; padding: 1px;}
+  div.bar {height: %dpx; background-color: green;}
+  </style>
+</head>
+<body>
+  <h2>Block coverage</h2>
+""" % (chart_width, chart_row_height)
+  FOOTER = """</body>
+</html>
+"""
+
+  out_file.write(HEADER)
+  if msg:
+    out_file.write('  <p>')
+    out_file.write(msg)
+    out_file.write('\n')
+  block_data.sort()
+  out_file.write(
+      '  <table>\n'
+      '    <tr><th colspan=3>&nbsp;')
+  for name in names:
+    out_file.write('<th colspan=3>%s' % name)
+  out_file.write('\n')
+  out_file.write(
+      '    <tr><th>Range<th>Block Name<th>Count')
+  for _ in names:
+    out_file.write('<th>Count<th>Pct<th>Chart')
+  out_file.write('\n')
+
+  max_block_size = max(len(t[3]) for t in block_data)
+  if max_block_size > cp_limit:
+    max_block_size = cp_limit
+  for start, end, name, block_cps, block_covered_cps_list in block_data:
+    range_str = '%04x-%04x' % (start, end)
+    num_in_block = len(block_cps)
+    tot_ht = 1 + int(math.floor(float(num_in_block) / cp_limit))
+    tot_px = int(float(chart_width) * num_in_block / max_block_size / tot_ht)
+    vir_cp_limit = int(float(num_in_block) / tot_ht)
+    out_file.write('    <tr>')
+    out_file.write('<td class="range">')
+    out_file.write(range_str)
+    out_file.write('<td class="name">')
+    out_file.write(name)
+    out_file.write('<td class="count">')
+    out_file.write(str(num_in_block))
+    for block_covered_cps in block_covered_cps_list:
+      num_covered = len(block_covered_cps)
+      pct = '%d%%' % int(100.0 * num_covered / num_in_block)
+      bar_ht = min(
+          tot_ht, 1 + int(math.floor(float(num_covered) / vir_cp_limit)))
+      bar_px = int(float(chart_width) * num_covered / max_block_size / bar_ht)
+      out_file.write('<td class="count">')
+      out_file.write(str(num_covered))
+      out_file.write('<td class="pct">')
+      out_file.write(pct)
+      out_file.write('<td><div class="chart">')
```

Comment:
Same comment as above. Though I guess with the stuff right below here it's a little hairy either way.